collector/filesystem_linux.go

```diff
@@ -118,28 +120,49 @@ func (c *filesystemCollector) processStat(labels filesystemLabels) filesystemSta
 		}
 	}
 
-	success := make(chan struct{})
-	go stuckMountWatcher(labels.mountPoint, success, c.logger)
-
 	buf := new(unix.Statfs_t)
-	err := unix.Statfs(rootfsFilePath(labels.mountPoint), buf)
-	stuckMountsMtx.Lock()
-	close(success)
-
-	// If the mount has been marked as stuck, unmark it and log it's recovery.
-	if _, ok := stuckMounts[labels.mountPoint]; ok {
-		level.Debug(c.logger).Log("msg", "Mount point has recovered, monitoring will resume", "mountpoint", labels.mountPoint)
-		delete(stuckMounts, labels.mountPoint)
+
+	// Allow mocking `unix.Statfs` values for testing.
```

Comment:
Wondering if there is a better way that avoids adding this test-only field..

Reply:
Hmm, given that this cannot be populated externally, I believe this shouldn't have any side-effects. I've also made some nits in the latest commit to explicitly denote the nature of the field.

LMK if you think there's a better alternative that would fit here, thanks!